### PR TITLE
Use Kotlin official code style

### LINKED ISSUE #17704: Use official Kotlin code style conventions
There were some internal discussions regarding code style and current configurations in this repository (paqN3M-OJ-p2) and there was an overall feeling that we should focus on using the official **Kotlin** [code style](https://kotlinlang.org/docs/coding-conventions.html) as close as possible and _maybe_ stop committing IDE-specific code style configs to Git, as files such as `.idea/codeStyles/Project.xml`.

That file frequently has its content updated because of IDE changes but is rarely updated. Besides that, having that file in the repo also makes us use outdated conventions. Another option, if we decide to keep the `codeStyles` files would be to make sure we maintain those code styles configs up-to-date.

The initial plan is to get this repo closer to what [WooCommerce](https://github.com/woocommerce/woocommerce-android) has in terms of code-styling, which allows us to always use Android Studio official updated conventions, and avoid conflicts in IDE config files.

Tasks:
- [ ] Update `.editorconfig` to keep some required settings (if needed)
- [ ] Remove Kotlin code-style settings from `.idea/codeStyles/Project.xml` or set it to the official defaults (`KOTLIN_OFFICIAL` code style defaults option in `Project.xml`)
- [ ] Reformat ALL Kotlin files in the project according to the up-to-date official code style in IDE
- [ ] Update `coding-style.md` with some instructions for manual IDE configurations (if needed)

Ideally, we would also completely remove the `Project.xml` file and migrate some configs to `.editorconfig` for other languages (`XML` and `Java`) BUT I was playing with that for a bit and that will introduce quite some different formatting (or just inflate the `.editorconfig` file with IntelliJ specific rules) so it's something to be discussed a bit more.

### PR DESCRIPTION
**Pull request by thomashorta**
Fixes #17704

Remove Kotlin-specific configuration in the `.idea/codeStyles/Project.xml` file in favor of using the default Kotlin official style conventions:

```
<option name="CODE_STYLE_DEFAULTS" value="KOTLIN_OFFICIAL" />
```

Also, reformat all Kotlin files according to this new style to ensure consistency after the above changes.

Most of the changes in this PR is related to the reformatting of .kt files, but the important changes related to the code styles and checks that need revision are in the following files:
- `.idea/codeStylesProject.xml`
- `config/checkstyle.xml` - changes here to make sure the `@Inject` rules target only Java files

**IMPORTANT: after pulling these changes locally I recommend CLOSING Android Studio completely (all windows) and opening it again, as changes in `Project.xml` seem to have some bugs syncing with the IDE.**

_NOTE: this PR is huge, opening the "Files changed" GitHub tab in the browser MIGHT cause the tab to crash or stop responding._

To test:
N/A

## Regression Notes
1. Potential unintended areas of impact
App compilation errors, linting errors, or runtime errors.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Limited manual testing (change affects _only_ formatting, should not introduce behavior changes)

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by ravishanker — APPROVED** [review]
Thanks for undertaking this 👍

**Comment by thomashorta** [comment]
I just synced this branch with `trunk` and reformatted everything again.

One thing I noticed is that empty lines were added between annotated properties this time. After some investigation, it seems Android Studio is unable to do all formatting at once so before when I ran reformat once, it just added a line break after each annotation, and this time it added blank lines between annotated properties, which is also part of the convention.

With that in mind, I ran reformatting twice, so formatting is properly applied on top of the most recent changes in `trunk`.

---

### Practical example

#### Before
```kotlin
@Inject lateinit var dispatcher: Dispatcher
@Inject lateinit var accountStore: AccountStore
@Inject lateinit var siteStore: SiteStore
```

#### After the first reformat (before today's changes)
```kotlin
@Inject
lateinit var dispatcher: Dispatcher
@Inject
lateinit var accountStore: AccountStore
@Inject
lateinit var siteStore: SiteStore
```

#### After the second reformat (after today's changes)
```kotlin
@Inject
lateinit var dispatcher: Dispatcher

@Inject
lateinit var accountStore: AccountStore

@Inject
lateinit var siteStore: SiteStore
```

**Review by mokagio — APPROVED** [review]
I just wanted to endorse the idea of using the official style guide, in particular for an open source project such as this WordPress Android. I feel like it will remove the need to discuss personal style preferences among existing contributors and help new ones get started.

File: .idea/codeStyles/Project.xml
Comment (irfano) [inline .idea/codeStyles/Project.xml:227]: Do we have to specify `CODE_STYLE_DEFAULTS` as `KOTLIN_OFFICIAL`? If we don't define `CODE_STYLE_DEFAULTS`, isn't the default value `KOTLIN_OFFICIAL` already?

**Review by irfano — APPROVED** [review]
Thank you very much for dealing with this. ❤️ 
I'd say we don't need to be afraid of the vast amount of changed files since they are only style changes. Before, I only had issues with reformatting when I did it for XML files. Sometimes it changes the order of elements, and that causes changes in the layout. But we're not updating XML files here, so it looks safe. 👍🏻

File: .idea/codeStyles/Project.xml
Comment (thomashorta) [reply .idea/codeStyles/Project.xml:227]: That was my first thought as well, but I found that if we don't have the CODE_STYLE_DEFAULTS set explicitly to `KOTLIN_OFFICIAL` then AS uses its own defaults (which seem to match "Kotlin obsolete IntelliJ IDEA codestyle" from the CodeStyle settings).

image

See this video I made checking the `Project.xml` on a new project and changing codestyle settings. _Note: for better immersion while watching the video, imagine myself shrugging in disbelief at the end of it!_

https://user-images.githubusercontent.com/5091503/210786489-92d56cd5-3f54-45e4-954f-ce898cb3d8a3.mov

**Comment by thomashorta** [comment]
I know this has a few approvals already but I will wait for @ParaskP7 to take a look as well as he did raise some codestyle discussions in the past and had some points as well when I proposed this change. I believe he will be able to take a look next week.

Until then I will keep syncing this PR with `trunk` daily so it doesn't get too behind in terms of changes and conflicts.

File: .idea/codeStyles/Project.xml
Comment (irfano) [reply .idea/codeStyles/Project.xml:227]: I understand. I assumed Android Studio would use the most up-to-date style by default. Apparently, it uses an obsolete codestyle instead. Thanks for the clarification.

**Comment by ParaskP7** [comment]
👋 @thomashorta and happy new year! 🎉 

> I know this has a few approvals already but I will wait for @ParaskP7 to take a look as well as he did raise some codestyle discussions in the past and had some points as well when I proposed this change.

Hey, please don't be blocked by me on that, my code style related points were mainly on the version control side of things and the `.idea` files. This change is trying to deal with a different problem and I am all-in on it no matter, so this is a big 🟢 from me, thank you! 🥇 

> I believe he will be able to take a look next week.

I did take a quick look, just to unblock you so you could merge this PR as soon as possible. I mainly looked at the below commits and some random file changes just to make sure that the indentation is the expected one:
- [keep relevant Kotlin configs in Project.xml and set defaults](https://github.com/wordpress-mobile/WordPress-Android/commit/883398f892c933fc002d07cedb126a1803176e59)
- [remove XML section from Projects.xml to match recent AS configs](https://github.com/wordpress-mobile/WordPress-Android/commit/84fcdfefed82c443fb86457e7103c3086c3bdf95)
- [update some checkstyle.xml rules to target only java](https://github.com/wordpress-mobile/WordPress-Android/commit/87c5892071b1d5e930407cf067539c3380ca1980)

Everything looks good to me, many thanks for this update! 💯 

> Until then I will keep syncing this PR with trunk daily so it doesn't get too behind in terms of changes and conflicts.

I hope this is the last time you sync this PR with `trunk`, let's do this! 🤞 😅 🟢

**Pull request merged**
